### PR TITLE
assistant builder retrieval config

### PR DESCRIPTION
**Pull request by fontanierh**
This PR implements a lot of the "retrieval config" of the assistant builder. There is still a lot of functionality and polish missing, but the PR is already big enough as-is

[Screenshots are outdated, the best would be to run the branch directly]

### CODE REVIEW

File: front/components/AssistantBuilderDataSourceModal.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  useEffect(() => {
+    if (dataSourceToManage) {
+      setSelectedDataSource(dataSourceToManage.dataSource);
+      setSelectedResources(dataSourceToManage.selectedResources);
+    }
+  }, [dataSourceToManage]);
+
+  const onClose = () => {
+    setOpen(false);
+    setTimeout(() => {
```
Comment (fontanierh) [inline front/components/AssistantBuilderDataSourceModal.tsx:58]: Timeout is useful so you don't see the modal's state reset during the modal close animation

File: front/components/DataSourceResourceSelectorTree.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
+import {
```
Comment (fontanierh) [inline front/components/DataSourceResourceSelectorTree.tsx:1]: Will require some DRY-ing between this and the Permission tree

**Review by PopDaph — DISMISSED** [review]
Following IRL discussion,  some css classed in components can be cleaned up a bit to simplify and there is a behavior to probably change for the bin icon button, but nothing prevents merging and fixing in other PRs. Great split of component that makes it easier to understand what we're doing 👍🏻

File: front/lib/connector_providers.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { DriveLogo, GithubLogo, NotionLogo, SlackLogo } from "@dust-tt/sparkle";
```
Comment (spolu) [inline front/lib/connector_providers.ts:1]: Thank you :100:

File: front/components/AssistantBuilderDataSourceModal.tsx [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+                  dataSource.connectorProvider as ConnectorProvider
+                ]?.isNested
+              }
+              selectedParentIds={new Set(Object.keys(selectedResources))}
+              onSelectChange={onSelectChange}
+            />
+          </div>
+          <div className="sticky top-16 hidden h-full flex-col md:flex">
+            <div className="flex flex-row">
+              {onDelete && (
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/AssistantBuilderDataSourceModal.tsx:235]: Agreed that delete button is weirdly placed.

But we will nuke that list likely and move the delete button in a better spot.

**Review by spolu — APPROVED** [review]
LGTM

Nice work! We'll likely nuke some of it :grimacing: but great progress.

What's the plan for "I want all that data source" ?

**Comment by fontanierh** [comment]
> What's the plan for "I want all that data source" ?

My understanding is that we decided not to do it ? Can add if we want to revisit the decision though

**Comment by spolu** [comment]
We decided to not support « all DataSources » but we can surely support in the api an entire data source (no parent filter) ?

**Comment by fontanierh** [comment]
> We decided to not support « all DataSources » but we can surely support in the api an entire data source (no parent filter) ?

Will look into adding it 👍

**Pull request merged**